src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/utils/UserMdcFilter.kt

```diff
@@ -9,19 +10,21 @@ import org.slf4j.MDC
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.core.annotation.Order
 import org.springframework.stereotype.Component
-import uk.gov.justice.digital.hmpps.incidentreporting.config.AuthenticationFacade
+import uk.gov.justice.hmpps.kotlin.auth.HmppsAuthenticationHolder
 import java.io.IOException
 
 @Component
 @Order(1)
-class UserMdcFilter @Autowired constructor(private val authenticationFacade: AuthenticationFacade) : Filter {
+class UserMdcFilter @Autowired constructor(
```

Comment:
I've opted to remove this filter in some of our classes, since it is only for `AppTraces` logging, `AppRequests` and `AppEvents` uses the [ClientTrackingConfiguration](https://github.com/ministryofjustice/hmpps-incident-reporting-api/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/config/ClientTrackingConfiguration.kt#L36) instead.

Reply:
I've got a task to add the `ClientTrackingConfiguration` to the kotlin lib as well, but need to spend some time to work out if I can use the `HmppsAuthenticationHolder` in it as all existing implementations

Reply:
i saw that you removed it, but didn't know what it does in the first place so was more cautious :p

on that note, i recall that you also removed `JwtAuthInterceptor`, but kept `UserContext` and `UserContextFilter` – should they all be dropped? it looked like the user context stuff was only referred to inside the auth interceptor. (again i have no idea what they are _supposed_ to do so couldn't check!)

Reply:
I think in your case they can all be dropped.  Some uses of the `UserContext` are to grab the authentication out of the request to then set for subsequent calls e.g. https://github.com/ministryofjustice/cmd-api/blob/de106d425ed822c6c6fbc1fcd4a5885f1170e2c9/src/main/java/uk/gov/justice/digital/hmpps/cmd/api/config/WebClientConfiguration.kt#L82, but can't see that being used here